### PR TITLE
Code Improvements and Clean Up

### PR DESCRIPTION
**Pull request by rafaelsc**
- Use async Dispose for File Generation
- Use read-only property when possible.
- Code CleanUp and Simplification
- Better lambda variable naming. 
- Fix Typos
- Update Test Instructions

### CODE REVIEW

File: src/TypedSignalR.Client.TypeScript/SpecialSymbols.cs [hunk 1]
```diff
@@ -20,23 +20,8 @@ public SpecialSymbols(Compilation compilation)
...
-
-        if (transpilationSourceAttributeSymbol is null)
-        {
-            throw new InvalidOperationException("Tapper.TranspilationSourceAttribute is not found");
-        }
-
-        HubAttributeSymbol = hubAttributeSymbol;
-        ReceiverAttributeSymbol = receiverAttributeSymbol;
-        TranspilationSourceAttributeSymbol = transpilationSourceAttributeSymbol;
+        HubAttributeSymbol = hubAttributeSymbol ?? throw new InvalidOperationException("TypedSignalR.Client.HubAttribute is not found");
```
Comment (nenoNaninu) [inline src/TypedSignalR.Client.TypeScript/SpecialSymbols.cs:23]: Don't use a null-coalescing operator to throw Exception.

File: tests/README.md [hunk 2]
```diff
@@ -7,9 +7,9 @@ $ dotnet run --project ./tests/TypedSignalR.Client.TypeScript.Tests.Server/Typed
 ```
 
 
-Generate TypeScript code.
+Generate TypeScript code. From last generator code.
 ```
-$ dotnet tsrts --project ./tests/TypedSignalR.Client.TypeScript.Tests.Shared/TypedSignalR.Client.TypeScript.Tests.Shared.csproj --out ./tests/TypeScriptTests/src/generated
+$  dotnet run --project ./src/TypedSignalR.Client.TypeScript.Generator/TypedSignalR.Client.TypeScript.Generator.csproj -- -p ./tests/TypedSignalR.Client.TypeScript.Tests.Shared/TypedSignalR.Client.TypeScript.Tests.Shared.csproj -o ./tests/TypeScriptTests/src/generated
```
Comment (nenoNaninu) [inline tests/README.md:12]: Inappropriate.

File: src/TypedSignalR.Client.TypeScript/SpecialSymbols.cs (hunk 1 above)
Comment (rafaelsc) [reply src/TypedSignalR.Client.TypeScript/SpecialSymbols.cs]: Reverted

File: tests/README.md (hunk 2 above)
Comment (rafaelsc) [reply tests/README.md]: Reverted

**Comment by nenoNaninu** [comment]
thanks!

**Pull request merged**